src/zenml/integrations/daytona/flavors/daytona_orchestrator_flavor.py

```diff
@@ -0,0 +1,179 @@
+#  Copyright (c) ZenML GmbH 2025. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Daytona orchestrator flavor."""
+
+from typing import TYPE_CHECKING, Dict, List, Optional, Type
+
+from zenml.config.base_settings import BaseSettings
+from zenml.integrations.daytona import DAYTONA_ORCHESTRATOR_FLAVOR
+from zenml.orchestrators import BaseOrchestratorConfig
+from zenml.orchestrators.base_orchestrator import BaseOrchestratorFlavor
+from zenml.utils.secret_utils import SecretField
+
+if TYPE_CHECKING:
+    from zenml.integrations.daytona.orchestrators.daytona_orchestrator import (
+        DaytonaOrchestrator,
+    )
+
+
+class DaytonaOrchestratorSettings(BaseSettings):
+    """Settings for the Daytona orchestrator.
+
+    Attributes:
+        api_key: The API key for authenticating with Daytona.
+        server_url: The URL of the Daytona server.
+        target: The target environment for Daytona.
+        custom_commands: Custom commands to run in the workspace.
+        synchronous: If True, wait for the pipeline to complete. If False,
+            return immediately after starting the pipeline. Defaults to False.
+        image: Custom Docker image to use for the workspace.
+        os_user: Operating system user for the workspace.
+        env_vars: Additional environment variables to set in the workspace.
+        labels: Labels to attach to the workspace.
+        public: Whether the workspace should be public.
+        cpu: Number of CPUs to allocate to the workspace.
+        memory: Memory in MB to allocate to the workspace.
+        disk: Disk size in GB to allocate to the workspace.
+        gpu: Number of GPUs to allocate to the workspace.
+        timeout: Timeout in seconds for workspace operations.
+        auto_stop_interval: Interval in seconds after which to automatically stop the workspace.
+    """
+
+    # Authentication and connection settings
+    api_key: Optional[str] = SecretField(default=None)
+    server_url: Optional[str] = "https://daytona.work/api"
```

Comment:
The convention is that stuff that defines the infrastructure like the `server_url` should be in the config and not be able to set at runtime, which does not respect RBAC permissions.